### PR TITLE
Template based protobuf parameters for ssz configurations

### PR DESCRIPTION
**Pull request by prestonvanloon**
This PR adds the ability to perform variable substitution on the ssz tags in protobufs. 

TL;DR build with `--define ssz=mainnet` (default value) for mainnet ssz config hard coded in the protos or build with `--define ssz=minimal` to for minimal ssz tag configs in protos.

Here's an example of how we wrap the proto files with a custom ssz rule which has a `config` argument to determine which set of variable substitution to use. 

```python
ssz_proto_files(
    name = "ssz_proto_files",
    srcs = [
        "messages.proto",
        "types.proto",
    ],
    config = select({
        "//conditions:default": "mainnet",
        "//proto:ssz_mainnet": "mainnet", # --define ssz=mainnet
        "//proto:ssz_minimal": "minimal", # --define ssz=minimal
    }),
)

proto_library(
    name = "v1_proto",
    srcs = [":ssz_proto_files"], # Template substituted files are provided as the sources.
    visibility = ["//visibility:public"],
    deps = [
        "//proto/eth/v1alpha1:v1alpha1_proto",
        "@com_google_protobuf//:timestamp_proto",
        "@gogo_special_proto//github.com/gogo/protobuf/gogoproto",
    ],
)
```

The substitutions are defined in proto/ssz_proto_library.bzl
```python
mainnet = {
    "block_roots.size": "8192,32",
    "state_roots.size": "8192,32",
    "eth1_data_votes.size": "1024",
    "randao_mixes.size": "65536,32",
    "active_index_roots.size": "65536,32",
    "compact_committees_roots.size": "65536,32",
    "previous_epoch_attestations.max": "8192",
    "current_epoch_attestations.max": "8192",
    "previous_crosslinks.size": "1024",
    "current_crosslinks.size": "1024",
    "slashings.size": "8192",
}

minimal = {
    "block_roots.size": "64,32",
    "state_roots.size": "64,32",
    "eth1_data_votes.size": "16",
    "randao_mixes.size": "64,32",
    "active_index_roots.size": "64,32",
    "compact_committees_roots.size": "64,32",
    "previous_epoch_attestations.max": "1024",
    "current_epoch_attestations.max": "1024",
    "previous_crosslinks.size": "8",
    "current_crosslinks.size": "8",
    "slashings.size": "64",
}
```

So when we have a proto file provided with `ssz_proto_files` and has any of those variables, they will be substituted at build time. 

Example
```proto
// In the proto file
repeated bytes block_roots = 2002 [(gogoproto.moretags) = "ssz-size:\"block_roots.size\""];

// With --define ssz=mainnet (default value), the variable is substituted.
repeated bytes block_roots = 2002 [(gogoproto.moretags) = "ssz-size:\"8192,32\""];

// With --define ssz=minimal.
repeated bytes block_roots = 2002 [(gogoproto.moretags) = "ssz-size:\"64,32\""];
```

This PR also breaks out minimal and mainnet spectests into their own targets. Eventually, the minimal targets will be able to specify that they want their dependencies built with `ssz=minimal`. Those tests that fail with mainnet prots are disabled until #3066.

### CODE REVIEW

**Comment by terencechain** [comment]
Woohoo!
